lib/view_model/exchange/exchange_view_model.dart

```diff
@@ -446,8 +455,12 @@ abstract class ExchangeViewModelBase extends WalletChangeListenerViewModel with
         if (!providersForCurrentPair().contains(provider)) continue;
 
         try {
-          final tempLimits =
-              await provider.fetchLimits(from: from, to: to, isFixedRateMode: isFixedRateMode);
+          final tempLimits = await provider
+              .fetchLimits(from: from, to: to, isFixedRateMode: isFixedRateMode)
+              .timeout(
+                Duration(seconds: 7),
+                onTimeout: () => Limits(max: 0.0, min: double.maxFinite),
+              );
```

Comment:
that's still in a for loop, so it will wait for each one until it returns then it will proceed to the next.

make it like the fetch rate in a Future.wait so all of them are called together